sdk/communication/Azure.Communication.CallAutomation/src/Models/RecognizeResultType.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Communication.CallAutomation.Models
+{
+    /// <summary>
+    /// Determines the sub-type of the recognize result.
+    /// </summary>
+    public readonly partial struct RecognizeResultType: IEquatable<RecognizeResultType>
```

Comment:
Any reason that we cannot use enum here?

Reply:
Do we already know all the possible values this RecognizeResultType could get? If there is a chance that in the future there will be more types, we have to use the struct.